spec/libraries/course/assessment/programming_test_case_report_spec.rb

```diff
@@ -202,6 +202,124 @@
     end
   end
 
+  # Test dynamic hints, failure and error messages, messages hash,
+  # output meta
+  context 'when given a report with various kinds of output data' do
+    let(:report_path) do
+      File.join(Rails.root, 'spec/fixtures/course/'\
+                'programming_messages_test_report.xml')
+    end
+
+    let(:report_xml) { File.read(report_path) }
+
+    let(:parsed_report) do
+      Course::Assessment::ProgrammingTestCaseReport.new(report_xml)
+    end
+    let(:public_test_cases) { parsed_report.test_suites.first.test_cases }
+    let(:private_test_cases) { parsed_report.test_suites.second.test_cases }
+
+    describe Course::Assessment::ProgrammingTestCaseReport::TestCase do
+      describe 'failed test case' do
+        subject { public_test_cases.first } # passed test case with output meta
+
+        describe '#output' do
+          it 'returns the output attribute' do
+            expect(subject.output).to eq('-1')
+          end
+        end
+
+        describe '#messages' do
+          it 'returns the output and failure message in a hash' do
+            expected_failure_contents = "\n<![CDATA[Traceback (most recent call last):\n  "\
```

Comment:
After you change the report, this should be a sample message, currently it's hard for readable and modify...